.github/workflows/ci-libshortfin.yml

```diff
@@ -152,3 +146,23 @@ jobs:
       run: |
         ctest --timeout 30 --output-on-failure --test-dir build
         pytest -s --durations=10
+
+  # Depends on all other jobs to provide an aggregate job status.
+  ci_libshortfin_summary:
+    if: always()
+    runs-on: ubuntu-24.04
+    needs:
+      - build-and-test
+    steps:
+      - name: Getting failed jobs
+        run: |
+          echo '${{ toJson(needs) }}'
+          FAILED_JOBS="$(echo '${{ toJson(needs) }}' \
+            | jq --raw-output \
+            'map_values(select(.result!="success" and .result!="skipped")) | keys | join(",")' \
+          )"
+          echo "failed-jobs=${FAILED_JOBS}" >> $GITHUB_OUTPUT
+          if [[ "${FAILED_JOBS}" != "" ]]; then
+            echo "The following jobs failed: ${FAILED_JOBS}"
+            exit 1
+          fi
```

Comment:
This failed jobs list code is forked from IREE. We could simplify for cases where there is only one job or we aren't using the `GITHUB_OUTPUT` (e.g. to send an alert to an external service).

Reply:
I don't know if there are plans to send it out via `GITHUB_OUTPUT` thus simplifying sounds good to me and if really needed we know where to fork the complexity from :smile:

Reply:
Okay, I'll simplify. Will need to test a bit :P

Reply:
Could just `echo "The job matrix failed"` I guess...

Reply:
Eh. Supporting a list of jobs is still useful for the future. I'll just remove the currently unused `echo "failed-jobs=${FAILED_JOBS}" >> $GITHUB_OUTPUT`.